cirqtools/qudit/common_gates.py

```diff
@@ -136,3 +138,49 @@ def _circuit_diagram_info_(self, args):
             return '[{}<->{}]'.format(self.flip_a, self.flip_b)
         return cirq.CircuitDiagramInfo((
             '[{}<->{}]'.format(self.flip_a, self.flip_b),))
+
+class Chrestenson(cirq.SingleQubitGate):
+    def __init__(self,dimension):
+        self.dimension = dimension
+
+    def _qid_shape_(self):
+        return (self.dimension,)
+
+    def _unitary_(self):
+        idx = np.arange(self.dimension, dtype=np.complex128)
+        u = idx[:, np.newaxis] * idx[np.newaxis, :]
+        u *= 2j*np.pi/self.dimension
+        np.exp(u, out=u)
+        u /= np.sqrt(self.dimension)
+        return u
+
+    def __pow__(self, exponent):
+        if exponent % 2 == 1:
+            return self
```

Comment:
My mistake on this.  Since only C^4 is identity, this should be
```python
if exponent % 4 == 1:  # power 1
    return self
if exponent % 4 == 3:  # power -1
    return InverseChrestenson(self.dimension)
```